inv/inventory.go

```diff
@@ -124,10 +124,31 @@ func (i *inventory) DeleteDevice(ctx context.Context, id model.DeviceID) error {
 }
 
 func (i *inventory) UpsertAttributes(ctx context.Context, id model.DeviceID, attrs model.DeviceAttributes) error {
-	if _, err := i.db.UpsertDevicesAttributes(
-		ctx, []model.DeviceID{id}, attrs,
-	); err != nil {
-		return errors.Wrap(err, "failed to upsert attributes in db")
+	needsUpsert := true
```

Comment:
Not critical; but you could also put this logic inside `UpsertDevicesAttributes` instead which also covers the internal endpoint for propagating attributes.